### DISCUSSION TITLE
My shot on cpp-template example with workers and with usage of Property helper

### DISCUSSION THREAD

**Discussion by DaMilyutin** [slint-ui/slint#620] [Show and tell] [answered]
I was going to have something like this
![image](https://user-images.githubusercontent.com/44639272/139355899-43dec77c-f2db-4226-8e49-38b69ec606b2.png)

And I managed to implement it! 
To use this code one has to clone original sixtyfps-cpp-template and replace files in folders with ones from zip.
[sixtyfps-cpp-template-with-workers.zip](https://github.com/sixtyfpsui/sixtyfps/files/7438125/sixtyfps-cpp-template-with-workers.zip)

I'd like to have any constructive feedback
1) On C++ Property helper (It can be done on code generator side as well)
2) On my usage of sixty fps functions and methods, maybe there are more optimal and less wordy ways.
2) On application and code

I also hope, that this example is useful for new developers.
Maybe, this can be added to some branch of cpp-template.

**Comment by tronical** [comment, marked as answer]
Hi! I looked a bit through your code. I think it looks fine :-)

Regarding the property helper, I can see how this seemingly makes things a lot easier. However I'm still not sure if hiding the threading entirely is the way to go, or if it should be more visible in the application design where data is exchanged between threads. I prefer the latter, where well defined channels are used to send minimal amounts of data across (programming with golang tainted me, but I not in a bad way :-).

I suggest we keep this here in the discussions and see what other feedback comes in. I suggest to put the code into a public git repo of yours and then we can link to it from here.

**Reply by DaMilyutin** [reply]
Thank you for feedback!
I'll fork cpp-template and provide a link.
Now I've added threadpool and have fixed some difficulties with it. The descussion for this case is created, too.

As to property helper, it explicitly requires to call `.blocking()` or `.safe()` wrappers when threading interactions.
Implicit calls with property itself are **NOT** thread safe.

**Reply by DaMilyutin** [reply]
@tronical , @ogoffart , I've added code with worker threads to my repository
https://github.com/DaMilyutin/sixtyfps-cpp-template-worker-threads
you may use it as you please.
Also, check Property.

Note, it is without thread pool since I'm still confused how to kill it when close button is pressed.
